packages/code-editor/src/CodeEditor/stories/Xml.tsx

```diff
@@ -0,0 +1,322 @@
+import { forwardRef, useRef, useState } from "react";
+import { css } from "@emotion/css";
+import {
+  HvCodeEditor,
+  HvCodeEditorProps,
+} from "@hitachivantara/uikit-react-code-editor";
+import {
+  HvButton,
+  HvButtonProps,
+  HvDialog,
+  HvDialogContent,
+  HvDialogTitle,
+  HvTooltip,
+  HvTreeItem,
+  HvTreeItemProps,
+  HvTreeView,
+  HvTypography,
+  theme,
+  uniqueId,
+} from "@hitachivantara/uikit-react-core";
+import { Code, Info } from "@hitachivantara/uikit-react-icons";
+
+const xsdSchema = `<?xml version="1.0" encoding="UTF-8"?>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
+           elementFormDefault="qualified"
+           attributeFormDefault="unqualified">
+  <xs:element name="library">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:element name="book" maxOccurs="unbounded">
+          <xs:complexType>
+            <xs:sequence>
+              <xs:element name="title" type="xs:string" minOccurs="1" />
+              <xs:element name="author" type="xs:string" minOccurs="1" />
+              <xs:element name="year" type="xs:integer" minOccurs="0" />
+              <xs:element name="genre" type="xs:string" minOccurs="0" />
+            </xs:sequence>
+            <xs:attribute name="isbn" type="xs:string" use="required" />
+          </xs:complexType>
+        </xs:element>
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+</xs:schema>`;
+
+const defaultValue = `<library>
+  <book isbn="123-0-456-0-789">
+    <title>Demon Copperhead</title>
+    <author>Barbara Kingsolver</author>
+    <year>2022</year>
+    <genre>Historical Fiction</genre>
+  </book>
+  <book isbn="987-0-654-0-321">
+    <title>Assassin's Apprentice</title>
+    <author>Robin Hobb</author>
+    <year>1995</year>
+    <genre>Fantasy</genre>
+  </book>
+</library>`;
+
+const classes = {
+  headerRoot: css({
+    display: "flex",
+    justifyContent: "space-between",
+    alignItems: "center",
+    border: `1px solid ${theme.colors.atmo4}`,
+    borderBottom: "none",
+    background: theme.colors.atmo1,
+    padding: theme.spacing("xs", "sm"),
+  }),
+  actionContainer: css({
+    display: "flex",
+    alignItems: "center",
+    gap: theme.space.xs,
+  }),
+  label: css({ display: "flex", alignItems: "center" }),
+};
+
+const Header = ({
+  onClickSearch,
+  onClickTree,
+}: {
+  onClickSearch: HvButtonProps["onClick"];
+  onClickTree: HvButtonProps["onClick"];
+}) => (
+  <div className={classes.headerRoot}>
+    <div className={classes.actionContainer}>
+      <HvTypography variant="label">XML</HvTypography>
+      <Code />
+    </div>
+    <div className={classes.actionContainer}>
+      <HvButton variant="primaryGhost" onClick={onClickSearch}>
+        Search
+      </HvButton>
+      <HvButton variant="primaryGhost" onClick={onClickTree}>
+        XML Tree
+      </HvButton>
+    </div>
```

Comment:
We could simplify this with a `flex: 1` "separator - it's what we do in our ActionBar samples (+ the `gap`)
```suggestion
    <HvTypography variant="label">XML</HvTypography>
    <Code />
    <div style={{ flex: 1 }} />
    <HvButton variant="primaryGhost" onClick={onClickSearch}>
      Search
    </HvButton>
    <HvButton variant="primaryGhost" onClick={onClickTree}>
      XML Tree
    </HvButton>
```

Reply:
I'll update 🙏